### PR TITLE
delta.py added

### PR DESCRIPTION
**Pull request by Toku11**
Delta computation added

### CODE REVIEW

**Comment by keunwoochoi** [comment]
Hi, overall it looks good to me. Can I ask you two more things though? A unit test and a version up to `0.1.8` would be needed before merging it.

**Comment by Toku11** [comment]
Tests and version up added

File: kapre/delta.py [hunk 1]
```diff
@@ -55,7 +64,6 @@ def call(self, x):
         kernel = K.arange(-n,n+1,1,dtype=K.floatx())
         kernel = K.reshape(kernel,(1,kernel.shape[-1],1,1))
         
-        #output = tf.nn.conv2d(x,kernel,1,'VALID',data_format='NHWC')
         x = K.conv2d(x,kernel,1,data_format='channels_last')/denom
```
Comment (keunwoochoi) [inline kapre/delta.py:67]: could you run anything like `autopep8` for the formatting?

Comment (keunwoochoi) [reply kapre/delta.py:67]: actually, as in https://github.com/keunwoochoi/kapre/blob/master/tox.ini#L34, you'll need to install `black` and run it 
```
black --line-length 100 --skip-string-normalization --target-version=py37 {toxinidir}/kapre
black --line-length 100 --skip-string-normalization --target-version=py37 {toxinidir}/tests
```
which will do automatic formatting.

File: setup.py [hunk 2]
```diff
@@ -14,6 +14,8 @@
         'numpy >= 1.8.0',
         'librosa >= 0.5',
         'tensorflow >= 1.15',
+	'torch',
```
Comment (keunwoochoi) [inline setup.py:17]: I think it's a little extra to ask the users to install `torch` and `torchaudio`. Also, these are only used for the test, so I'd rather put them at https://github.com/keunwoochoi/kapre/blob/master/tox.ini#L14 and Line 21.

File: tests/test_deltas.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import pytest
+import torch
```
Comment (keunwoochoi) [inline tests/test_deltas.py:2]: I also think it'd be nicer if we can completely get away from using torch/torchaudio. The equations at https://pytorch.org/audio/functional.html#compute-deltas is broken now, but..
i)  maybe after waiting for them to fix it, then implement the test on numpy according to the equation
ii) or, use https://librosa.github.io/librosa/generated/librosa.feature.delta.html (not 100% sure if they refer to the same operation but i believe they should)?

File: tests/test_deltas.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        else:
+            DS = S[0, 0]
+
+        DS_scale_librosa = (np.max(DS_librosa) - np.min(DS_librosa))
+        DS_dif_librosa = np.abs(DS - DS_librosa) / DS_scale_librosa
+
+        DS_scale_torch = (np.max(DS_torch) - np.min(DS_torch))
+        DS_dif_torch = np.abs(DS - DS_torch) / DS_scale_torch
+
+        assert np.mean(DS_dif_torch) < 0.001
```
Comment (keunwoochoi) [inline tests/test_deltas.py:59]: imo we should use [np.allclose](https://numpy.org/doc/1.17/reference/generated/numpy.allclose.html).

File: kapre/delta.py (hunk 1 above)
Comment (Toku11) [reply kapre/delta.py:67]: Ok I will do it

File: setup.py (hunk 2 above)
Comment (Toku11) [reply setup.py:17]: I know, in fact test works with torch and Librosa... I didn't want to put it on setup but I can remove that part if with Librosa is enough

**Comment by Toku11** [comment]
Changes Requested applied, Test Unit working, black executed however it gave format to another files with inappropriate format. Travis built but there is one core dumped error, I cannot find the problem.

File: tests/test_deltas.py (hunk 3 above)
Comment (Toku11) [reply tests/test_deltas.py]: Done

File: tests/test_deltas.py (hunk 4 above)
Comment (Toku11) [reply tests/test_deltas.py]: Done

File: kapre/delta.py (hunk 1 above)
Comment (Toku11) [reply kapre/delta.py]: Done

**Comment by Toku11** [comment]
Any update? Travis only returns out of memory with no explanation.

**Comment by keunwoochoi** [comment]
Hey, thanks for the fixes!
- `Black` didn't need to be part of this PR imo but also useful so I'll keep it. I thought of setting it up anyway. 
- The unit test looks good now. I'll make some changes and commit to this branch directly now on.
  - I'll remove the notebook. It is not really relevant to `Kapre` but to `Tensorflow` :)
- I don't know why Travis test passes on py3.6 but not on 3.7. It seems not relevant to this PR, I'll merge it as long as it passes on 3.6.

Thanks again!

**Comment by keunwoochoi** [comment]
FWIW - There would be also some minor changes like names of layers (`ComputeDeltas` --> maybe.. `Delta`) as well. So after it's merged, you might want to double check. But I won't change its behavior.

**Comment by Toku11** [comment]
Ok, no problem with changes, I created notebook because I will start working on kapre MelSpectrogram changes because I’m using it and I realize results are no consistent as mentioned so I’ll try to fix it. If you want to delete it, it’s perfect.  https://github.com/keunwoochoi/kapre/issues/61#issue-528618091

**Comment by keunwoochoi** [comment]
@Toku11 Do you see this check box? Could you check it if so?

**Comment by keunwoochoi** [comment]
Nevermind, I'm continuing this PR at #71. I think in that way, all the commit logs are still preserved!

**Pull request closed without merging**